A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js

```diff
@@ -1002,5 +1003,209 @@ class A320_Neo_SAI_PressureIndicator extends HTMLElement {
+        st_alt_txt.setAttribute("y", boxRow2 + txt_off + "%");
+        this.selfTestSVG.appendChild(st_alt_txt);
+
+        let st_tmr = document.createElementNS(Avionics.SVG.NS, "rect");
+        st_tmr.setAttribute("id", "TmrTest")
+        st_tmr.setAttribute("fill", "#afbb3a");
+        st_tmr.setAttribute("x", "30%");
+        st_tmr.setAttribute("y", boxRow3 + "%");
+        st_tmr.setAttribute("width", boxWidthInit + "%");
+        st_tmr.setAttribute("height", boxHeight + "%");
+        this.selfTestSVG.appendChild(st_tmr);
+
+        this.st_tmr_txt = document.createElementNS(Avionics.SVG.NS, "text");
+        this.st_tmr_txt.setAttribute("id", "TmrTestTxt")
+        this.st_tmr_txt.textContent = "INIT "+ Math.ceil(this.testTimer)  +"s";
+        this.st_tmr_txt.setAttribute("font", "Roboto");
+        this.st_tmr_txt.setAttribute("font-weight", "900");
+        this.st_tmr_txt.setAttribute("font-size", "42px");
+        this.st_tmr_txt.setAttribute("fill", "#1f242d");
+        this.st_tmr_txt.setAttribute("x", "31%");
+        this.st_tmr_txt.setAttribute("y", boxRow3 + txt_off + "%");
+        this.selfTestSVG.appendChild(this.st_tmr_txt);
+
+        let st_att = document.createElementNS(Avionics.SVG.NS, "rect");
+        st_att.setAttribute("id", "AttTest")
+        st_att.setAttribute("fill", "#afbb3a");
+        st_att.setAttribute("x", "36%");
+        st_att.setAttribute("y", boxRow1 + "%");
+        st_att.setAttribute("width", boxWidth + "%");
+        st_att.setAttribute("height", boxHeight + "%");
+        this.selfTestSVG.appendChild(st_att);
+
+        let st_att_txt = document.createElementNS(Avionics.SVG.NS, "text");
+        st_att_txt.setAttribute("id", "AttTestTxt")
+        st_att_txt.textContent = "ATT";
+        st_att_txt.setAttribute("font", "Roboto");
+        st_att_txt.setAttribute("font-weight", "900");
+        st_att_txt.setAttribute("font-size", "42px");
+        st_att_txt.setAttribute("fill", "#1f242d");
+        st_att_txt.setAttribute("x", "38%");
+        st_att_txt.setAttribute("y", boxRow1 + txt_off + "%");
+        this.selfTestSVG.appendChild(st_att_txt);
+
+    }
+    update(dTime) {
+        if (!this.complete) {
+            if (this.testTimer >= 0) {
+                this.testTimer -= dTime / 1000;
+            }
+
+            if (this.testTimer > 9) {
+                this.st_tmr_txt.textContent = "INIT " + Math.ceil(this.testTimer) + "s";
+            }
+            else {
+                this.st_tmr_txt.textContent = "INIT 0" + Math.ceil(this.testTimer) + "s";
+            }
+            if (this.testTimer <= 0) {
+                this.finishTest();
+            }
+        }
+    }
+
+    resetTimer() {
+        this.testTimer = Math.floor(Math.random() * 3) + this.start_time;
+        this.complete = false;
+        let element = document.querySelector("#SelfTestHider");
+        element.style.display = "none";
+        this.selfTestDiv.style.display = "block";
+    }
```

Comment:
can you make it so this doesn't touch the DOM every frame while no power? maybe have 3 states, OFF, TESTING, ON, and don't run this code if it's OFF, and set it to OFF at bottom of this function?

Reply:
What if there's a greater than comparison that checks if the current time (testTimer) is greater than the default starting time (start_time), that skips the logic if true. Would this be acceptable?

Reply:
This is the logic that is in the latest commit.